DIPs/DIP1xxx-rn.md

```diff
@@ -0,0 +1,801 @@
+* offers encapsulation
+
+The downside of this solution is that the user must do all the field copying by hand
+and every time a field is added to a struct, the copy constructor must be modified.
+However, this can be easily bypassed by D's introspection mechanisms. For example,
+this simple code may be used as a language idiom or library function:
+
+```D
+foreach (i, ref field; src.tupleof)
+    this.tupleof[i] = field;
+```
+
+As shown above, the single benefit of the postblit can be easily substituted with a
+few lines of code. On the other hand, to fix the limitations of the postblit it is
+required that more complexity is added for little to no benefit. In these circumstances,
+for the sake of uniformity and consistency, replacing the postblit constructor with a
+copy constructor is the reasonable thing to do.
+
+## Description
+
+This section discusses all the technical aspects regarding the semantics
+of the copy constructor.
+
+### Syntax
+
+A declaration is a copy constructor declaration if it is a constructor declaration that takes only one
+parameter by reference that is of the same type as `typeof(this)`. Declaring a copy constructor as
+mentioned has the advantage that the parser does not get modified at all, thus leaving the language grammar
+unchanged.
+
+```d
+import std.stdio;
+
+struct A
+{
+    this(ref A rhs) { writeln("x"); }     // copy constructor
+}
+
+void main()
+{
+    A a;
+    A b = a;        // calls copy constructor implicitly
+    A c = A(b);     // calls constructor explicitly
+}
+```
+
+The copy constructor may also be called explicitly since it is also a constructor.
+
+The argument to the copy constructor is passed by reference in order to avoid infinite recursion (passing by
+value would require a copy of the `struct` which would be made by calling the copy constructor, thus leading
+to an infinite chain of calls).
+
+The type qualifiers may be applied to the parameter of the copy constructor, but also to the function itself
+in order to facilitate the ability to describe copies between objects of different mutability levels. The type
+qualifiers are optional.
+
+### Semantics
+
+This section discusses all the aspects regarding the semantic analysis and interaction with other
+components of the copy constructor.
+
+#### Copy constructor and postblit cohabitation
+
+In order to assure a smooth transition from postblit to copy constructor, the following
+strategy is employed: if a `struct` defines a postblit (user-defined or generated) all
+copy constructor definitions will be ignored for that particular `struct` and the postblit
+is going to be used. Existing codebases that didn't use the postblit may start using the
+copy constructor without any problems, while codebases that did rely on the postblit may
+start writing new code using the copy constructor while removing the deprecated postblit
+from their code.
+
+Transitioning from using a postblit to using a copy constructor may simply be done by
```

Comment:
Transitioning from using a postblit to a copy constructor -> The transition from postblit to copy constructor

may simply be done -> may be simply achieved